In [1]:
import pandas as pd
import re

### LCK 경기 데이터중 담원이 포지션을 이동해 경기를 했던 적이있어서
### 데이터가 중복으로 들어가는 것을 확인

In [2]:
lck = pd.read_csv('projectData/LCK1.csv', encoding='cp949', index_col=0)
lck_po = pd.read_csv('projectData/LCK_PO.csv', index_col=0)


# # 쇼메이커, 캐니언 겹치는 정보 드랍해주기
lck1 = lck[lck['call_name'] == 'DK ShowMaker']
lck2 = lck[lck['call_name'] == 'DK Canyon']
a =lck1[lck1['line'] == 'AD'].index
b =lck2[lck2['line'] == 'MID'].index

lck = lck.drop(index=a)
lck = lck.drop(index=b)

### 중복된 데이터의 인덱스값을 드랍해주는 형태로 제거

In [3]:
# PO 데이터까지 포함하기위해 데이터를 합침
LCK =pd.concat([lck,lck_po])

In [4]:
LCK

call_name    player_name line champion  game win_rate  kill  death  \
0      KT Doran  최현준2000.07.22  TOP      리 신     7    42.9%   3.0    3.0   
1      KT Doran  최현준2000.07.22  TOP      제이스     6    83.3%   3.7    2.3   
2      KT Doran  최현준2000.07.22  TOP      비에고     5    40.0%   2.6    1.8   
3      KT Doran  최현준2000.07.22  TOP      레넥톤     5    20.0%   2.0    3.2   
4      KT Doran  최현준2000.07.22  TOP       그웬     5    40.0%   3.2    2.8   
..          ...            ...  ...      ...   ...      ...   ...    ...   
249  LSB Effort  이상호2000.11.23  SUP       라칸     5    80.0%   0.2    3.2   
250  LSB Effort  이상호2000.11.23  SUP     노틸러스     1       0%   1.0    5.0   
251  LSB Effort  이상호2000.11.23  SUP      브라움     1       0%   0.0    3.0   
252  LSB Effort  이상호2000.11.23  SUP     탐 켄치     1       0%   0.0    2.0   
253  LSB Effort  이상호2000.11.23  SUP      카르마     1       0%   0.0    5.0   

     assist  kda  win  lose  dpm  
0       5.0  2.7    3     4  446  
1       5.8  4.1    5     1  628  
2       5.0  4.2    2     3  405  
3       3.8  1.8    1     4  401  
4       4.4  2.7    2     3  596  
..      ...  ...  ...   ...  ...  
249     7.6  2.4    4     1   96  
250     8.0  1.8    0     1  144  
251     3.0  1.0    0     1  128  
252     4.0  2.0    0     1   76  
253     4.0  0.8    0     1  184  

[761 rows x 13 columns]

## 다른 리그 데이터 모두 가져오기
### LPL과 VCS는 데이터가 없어서 제외

In [5]:
CBLOL = pd.read_csv('projectData/CBLOL1.csv',index_col='Unnamed: 0')
LCL = pd.read_csv('projectData/LCL1.csv',index_col='Unnamed: 0')
LCO = pd.read_csv('projectData/LCO1.csv',index_col='Unnamed: 0')
LCS = pd.read_csv('projectData/LCS1.csv',index_col='Unnamed: 0')
LEC = pd.read_csv('projectData/LEC1.csv',index_col='Unnamed: 0')
LJL = pd.read_csv('projectData/LJL1.csv',index_col='Unnamed: 0')
LLA = pd.read_csv('projectData/LLA1.csv',index_col='Unnamed: 0')
TCL = pd.read_csv('projectData/TCL1.csv',index_col='Unnamed: 0')
PCS = pd.read_csv('projectData/PCS1.csv',index_col='Unnamed: 0')

### 모든 리그 데이터를 합침

In [6]:
# 리그 데이터 합쳐주기
All_league = pd.concat([LCK,CBLOL,LCL,LCO,LCS,LEC,LJL,LLA,TCL,PCS])

In [7]:
# 중복값을 제거해 하나의 선수 이름값만 남김
All_league_player = All_league['player_name'].unique()
All_league_player

array(['최현준2000.07.22', '김동하1995.12.22', '김기인1999.05.28', '김광희1997.10.16',
       '노태윤2003.12.25', '이동주2001.03.08', '김창동2000.02.11', '박우태1998.12.08',
       '윤용호2000.08.25', '이재원1998.02.16', '박기태2001.09.26', '한왕호1998.02.03',
       '이진혁2000.06.07', '김건부2001.06.18', '김근성2000.02.09', '김태민1999.07.07',
       '문우찬1999.10.30', '박미르2000.05.24', '김정현2003.04.24', '강선구1998.04.30',
       '문현준2002.12.24', '엄성현1999.06.02', '김요한1999.12.17', '김민성2003.08.14',
       '김동범2000.02.23', '홍창현2000.03.12', '정지훈2001.03.03', '허수2000.07.22',
       '김재연1998.07.25', '유수혁2000.08.22', '김태훈1999.07.14', '곽보성1999.03.01',
       '황성훈2000.03.11', '이상혁1996.05.07', '김태우2000.08.09', '송수형2002.08.25',
       '송용준1996.02.25', '배호영2001.07.23', '박진성1998.03.15', '한겨레2000.01.12',
       '박재혁1998.12.29', '장용준1999.03.18', '이민형2002.02.06', '이채환2000.04.05',
       '서대길2000.02.25', '김혁규1996.10.23', '박정현1999.11.28', '오현택2001.10.04',
       '정현우2001.09.20', '김태윤2002.06.02', '박증환1999.10.08', '김형규2001.02.01',
       '조건희1997.04.01', '유

In [8]:
# 데이터를 담을 빈 리스트 작성
korean_player = []

# 반복문으로 리스트안에 각 데이터를 넣어줌
for i in All_league_player:
    korean_player.append(i)

In [9]:
# 한국어만 남기도록 처리해줌
korea = re.compile('[^ ㄱ-ㅣ가-힣]+')

# 한국어와 스플릿으로 구분된 빈칸 제거
result = korea.sub('',str(korean_player))
result2 = result.replace(' ','')
result2 = result2.replace("'",'')

# 한국인은 일반적으로 이름이 3글자니까 str 전체 길이에서 3으로 나누어줌
len(result2)/3

## 122.66666 인 이유는 허수라는 두글자이름이 있기때문에

122.66666666666667

## 전체 리그에서 활약하는 한국인 선수는 123명

## LCK만 확인해보기

In [10]:
## LCK는 모두 한국인 선수니까 그냥 중복값만 제거해서 확인

LCK_player = LCK['player_name'].unique()
len(LCK_player)

63

### 63명의 선수가 LCK에서 활약,  즉 123명의 선수중 63명은 LCK에있음

In [15]:
print(f'전체 선수의 수   : {(len(All_league_player))} ')
print(f'한국 선수의 비율 :  {round((123/len(All_league_player)) * 100)} % ')

전체 선수의 수   : 511 
한국 선수의 비율 :  24 % 


### 전체리그 511명의 선수 중 123명의 한국선수가 활약 중 
### 롤 프로게이머중 약 25%는 한국인 선수이다

In [17]:
# 메이저 리그만 확인하기
# 메이저 리그 데이터 합쳐주기
major_league = pd.concat([LCK,LEC,LCS])
major_league_player = major_league['player_name'].unique()

In [18]:
# 데이터를 담을 빈 리스트 작성
korean_player = []

# 반복문으로 리스트안에 각 데이터를 넣어줌
for i in major_league_player:
    korean_player.append(i)

In [19]:
# 한국어만 남기도록 처리해줌
korea = re.compile('[^ ㄱ-ㅣ가-힣]+')

# 한국어와 스플릿으로 구분된 빈칸 제거
result = korea.sub('',str(korean_player))
result2 = result.replace(' ','')
result2 = result2.replace("'",'')

# 한국인은 일반적으로 이름이 3글자니까 str 전체 길이에서 3으로 나누어줌
len(result2)/3

## 122.66666 인 이유는 허수라는 두글자이름이 있기때문에

71.66666666666667

In [20]:
result2

'최현준김동하김기인김광희노태윤이동주김창동박우태윤용호이재원박기태한왕호이진혁김건부김근성김태민문우찬박미르김정현강선구문현준엄성현김요한김민성김동범홍창현정지훈허수김재연유수혁김태훈곽보성황성훈이상혁김태우송수형송용준배호영박진성한겨레박재혁장용준이민형이채환서대길김혁규박정현오현택정현우김태윤박증환김형규조건희유환중윤세준손시우류민석오효성김정민장준수이상호이지융손민우신태민김찬호정언영허승훈나문용최재현조용인이동근한민국'

## 메이저리그 전체 한국인 수는 72명, LCK가 63명이니 나머지는 9명